garrysmod/gamemodes/base/entities/entities/base_trigger.lua

```diff
@@ -0,0 +1,95 @@
+
+ENT.Base = "base_entity"
+ENT.Type = "trigger"
+
+--[[---------------------------------------------------------
+	Name: Initialize
+-----------------------------------------------------------]]
+function ENT:Initialize()
+	self:SetTrigger(true)
+	self.Entities = {}
+end
+
+--[[---------------------------------------------------------
+	Name: IsTouchedBy
+-----------------------------------------------------------]]
+function ENT:IsTouchedBy(ent)
+	return self.Entities[ent]
```

Comment:
Well the check would return false anyway if the entity isn't valid and true otherwise